docs/internal/types.md

```diff
@@ -0,0 +1,236 @@
+
+The declaration is trivial:
+```ts
+// Helps receivers know they can convert the object to a string representation
+interface Serialisable
+{
+  // All derived classes will have the common field
+  var common : F64;
+
+  // Derived classes must implement this method
+  toString() : String;
+};
+
+class MyType : Serialisable
+{
+  // Inherits `common`
+  let field : I32;
+
+  // Some other method
+  innerSum(x : I32) : I32
+  {
+    let res : I32 = x + field;
+    res
+  }
+
+  // Implements `toString`
+  toString() : String
+  {
+    ...
+  }
+};
+```
+
+### Fields
+
+Define objects that belong to the class and will be instantiated (ie. use memory) on every instance of the class.
+
+Fields can be initialised in two ways:
+ * `var`: Must be initialised on construction, but can change after.
+ * `let`: Must be initialised on construction, and cannot change later.
+
+They can also be a private member, if its name starts with `_`.
+
+They can also have a keyword `embed`, which means ...
+
+### Methods
+
+Methods are functions that belong to a class or interface. Like functions, they can have arguments and a return type.
+
+Methods can be called statically (via `Class::method()`) or dynamically (via `object.method()`).
+On the static case, the `self` pointer is a descriptor forthe type, while on the dynamic case, it's a descriptor to the instance (and its vtable, type info, etc).
+
+Static calls cannot change the fields of the class unless an object is being created and returned, like in the case of constructors.
+
+### Constructors and Destructors
+
+Verona distinguishes memory allocation with object construction.
+There are no `constructors` per se, in Verona, only static methods that can allocate memory (using `new`), build an object and return it.
+
+Example:
+```ts
+class Capibara
+{
+  let name : String;
+  var weight : F32; // Kgs
+
+  // Create a baby capibara
+  newborn(name : String) : Capibara
+  {
+    var self : Capibara = new Capibara;
+    self.name = name;
+    self.weight = 3; // Tiny baby!
+    self; // Returns `self`
+  }
+
+  eat(self: Capibara, food : F32)
+  {
+    self.weight += food;
+  }
+
+  poop(self: Capibara)
+  {
+    self.weight -= 0.1; // Tiny poop!
+  }
+};
+```
+
+On the example above, the `newborn` method creates a `Capibara` with a specific name and a default weight.
+The `name` can't change, so we declare it with `let`, while the `weight` does change, so it's a `var`.
+
+There's nothing special about the method `newbord` in comparison with others, apart from the fact that it allocates, initialises and returns an object of the same type.
+
+Destructors are still under debate, but they won't have to deallocate memory like in C++, just execute finalization tasks like closing file handles, running some cleanup routines, checking final consistency, etc.
+
+## Primitive Types
+
+Verona treats numeric types specially in the compiler but their representaiton isn't special in the language.
```

Comment:
Typo: representaiton

Reply:
I *always* do that typo "ation" -> "aiton" because of how "ai" in PT reads like "a" in GB. :(

Reply:
```vim
:set spell
:hi SpellBad ctermfg=Red ctermbg=black cterm=undercurl
:set spelllang=en
```
:-)

Reply:
I don't like spell check on vim because it marks stupid things and I need to add a lot of stuff to the dictionary or the document looks like it's haemorrhaging. I usually run aspell from the terminal, but this time I forgot. :(